ADR/0008-integration-service-api-contracts.md

```diff
@@ -0,0 +1,266 @@
+
+Example output
+```
+[
+    {
+        "result": "FAILURE",
+        "namespace": "Red Hat App Studio E2E tests",
+        "timestamp": "1649148140",
+        "successes": 75,
+        "skipped": 14,
+        "note": "",
+        "failures": [
+            "[It] [release-suite test-demo] Creation of the 'Happy path' resources Create an ApplicationSnapshot."
+        ]
+    },
+    {
+        "result": "SUCCESS",
+        "timestamp": "1649843611",
+        "namespace": "Demo Tests",
+        "successes": 12,
+        "skipped": 0,
+        "note": "",
+        "failures": []
+    }
+]
+```
+
+#### Full Test Output JSON
+
+The Test Tekton tasks will use a standardized manner of displaying testing information. This information will be saved in the form of a JSON file. The contents of this file will relay results of validation by the [Open Policy Agent](https://www.openpolicyagent.org/) policies as executed by the [Conftest](https://www.conftest.dev/) tool and saved in its JSON output format.
+
+Each test will have a standardized short name in snake case, e.g. `release_label_required`, `architecture_label_required`, `architecture_label_deprecated` etc.
+
+The output will provide the following information about the overall test result:
+- **filename** - Name of the file that was inspected by the Tekton task
+  - This will reflect the data that the test was executed on, such as `image-inspect.json`, `clair-vulnerabilities.json` etc.
+- **namespace** - The rego namespace of the test policy. 
+- **successes** - The number of successful checks in the form of an integer
+- **failures** - A JSON list containing objects describing each failure
+
+The **failures** list of objects will provide the following information about each check:
+- **name** - Name of the individual check written in [snake case](https://en.wikipedia.org/wiki/Snake_case)
+- **msg** - Message about what went wrong with the check
+- **description** - The explanation about why the check is significant (Why the label is used, why the vulnerabilities are checked etc.)
+- **url** - link to the further documentation on the individual check (or that type of check)
+Example full JSON output for a Tekton task that tests the container image labels:
+```
+[
+    {
+        "filename": "image-inspect.json",
+        "namespace": "image_labels",
+        "successes": 19,
+        "failures": [
+            {
+                "msg": "The 'architecture' label is required",
+                "metadata": {
+                    "details": {
+                        "description": "Architecture the software in the image should target.",
+                        "name": "architecture_label_required",
+                        "url": "https://source.redhat.com/groups/public/container-build-system/container_build_system_wiki/guide_to_layered_image_build_service_osbs#jive_content_id_Labels"
+                    }
+                }
+            }
+          ]
+    }
+]
+```
+
+### Information injection
```

Comment:
This section is too open ended.

I don't think I see an example in [build-definitions](https://github.com/redhat-appstudio/build-definitions/) of a `ConfigMap` being used. I do see that `tasks/sast-snyk-check.yaml` uses a Secret.

Is the ConfigMap part relevant?

Reply:
Currently we use `Secrets` to inject configuration into Tasks only, will we use `ConfigMaps` next maybe?